x-pack/plugins/endpoint/server/routes/resolver/queries/children.ts

```diff
@@ -3,72 +3,92 @@
  * or more contributor license agreements. Licensed under the Elastic License;
  * you may not use this file except in compliance with the Elastic License.
  */
+import { SearchResponse } from 'elasticsearch';
+import { ResolverEvent } from '../../../../common/types';
 import { ResolverQuery } from './base';
+import { PaginationBuilder, PaginatedResults } from '../utils/pagination';
+import { JsonObject } from '../../../../../../../src/plugins/kibana_utils/public';
 
-export class ChildrenQuery extends ResolverQuery {
-  protected legacyQuery(endpointID: string, uniquePIDs: string[], index: string) {
-    const paginator = this.paginateBy('endgame.serial_event_id', 'endgame.unique_ppid');
+/**
+ * Builds a query for retrieving descendants of a node.
+ */
+export class ChildrenQuery extends ResolverQuery<PaginatedResults> {
```

Comment:
:grey_question: :cool: This `<PaginatedResults>` is what you meant before when you said "queries define their own pagination"?

Reply:
Yep, exactly. Each individual query can define what it's return type looks like instead of all of them having to prescribe to a single return type. The children query will have paginated results whereas the lifecycle query and stats, will not have paginated results and will have a slightly different return type.